app/models/alert_date.py

```diff
@@ -24,6 +24,18 @@ def date_as_lang(self):
         dt = self._local_datetime
         return f'{dt:%A} {dt.day} {dt:%B} {dt:%Y}'
 
+    @property
+    def as_url(self):
```

Comment:
get_url_for_alert is used to get a url from an alert - this is a property of the various timestamps within an alert. I don't think you should ever be using this in a context where you could confuse them.

That said, you shouldn't ever call this for a `cancelled_at_date` either, only a `starts_at_date`, so i'm not completely against moving this out of the `AlertDate` class and on to a property on the Alert object. It just felt neat on the AlertDate as it doesn't care about the rest of the alert object.

then it could be `Alert.start_date_for_url` or something like that perhaps? I think that my thought process when naming this function was that I'd like to keep reference to URL in the function definition just so it's obvious that it's got a bigger impact - i feel like we're generally aware that URLs are harder to change than other things. (Whereas a `short_date` for example someone may come along later and think "actually i prefer two digit year to four digit year" and cause irreversible damage).

Reply:
Ah I see that I had misunderstood. I thought the `as_url` was a property on the `Alert` rather than on the `AlertDate` class.

I am less worried now about what you currently have written and don't think it needs any changes.

Thanks for this.